src/main/java/uk/gov/digital/ho/pttg/application/util/MaxLengthNameNormalizer.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.digital.ho.pttg.application.util;
+
+import uk.gov.digital.ho.pttg.dto.Individual;
+
+public class MaxLengthNameNormalizer implements NameNormalizer {
```

Comment:
There is an awful lot of code in this solution when it could be achieved very simply - name = name.substring(0, MAX_NAME_LENGTH).  @SWhitlam Is this a situation where you would advocate simplicity over an elaborate design?

Reply:
You'd get an IndexOutOfBoundsException with this.

This object includes a null check and length check before this method is called to avoid this - but am always open to making my code cleaner / simpler if you have advice on this :)

Reply:
My school of thought was that we could inject all name rules together - the next being the special characters - to make the easily configurable & not to clutter the class.

This rule on its own I can see it being a bit overkill, so can move it into the class if you guys think it's best?

Reply:
OK maybe not that simple, but you get the point :)  A 6 line method would do the job.  I can only think of 2 normalisation requirements - length and special characters - so I'm not sure that requires a full blown solution?  I am still adjusting to the concept that quick and dirty code being better in some scenarios.

Reply:
It's not quick and dirty. It's writing enough code to meet the requirement and not overusing design patterns. As you say a 6 line method will fulfil this requirement. If we then get more requirements that mean a more complex solution is more appropriate we can refactor at that point rather than applying a solution we don't know we require.